code/modules/events/paradox_clone.dm

```diff
@@ -0,0 +1,75 @@
+/datum/event/paradox_clone
+	var/spawncount
+
+/datum/event/paradox_clone/setup()
+	spawncount = (round(length(get_living_players(exclude_nonhuman = FALSE, exclude_offstation = TRUE)) / 40))
+
+/datum/event/paradox_clone/proc/abort()
+	var/datum/event_container/EC = SSevents.event_containers[EVENT_LEVEL_MODERATE]
+	EC.next_event_time = world.time + 60 SECONDS // Fire again in a minute
+
+/datum/event/paradox_clone/start()
+	INVOKE_ASYNC(src, PROC_REF(wrapped_start))
+
+/datum/event/paradox_clone/proc/wrapped_start()
+	if(!spawncount) //if lower than 40 playas...
+		message_admins("Not enough players for Paradox Clone event: [length(get_living_players(exclude_nonhuman = FALSE, exclude_offstation = TRUE))]/40! Aborting. Choosing another moderate event.")
+		abort()
+		return
+
+	var/count = spawncount
+	var/wait_time = 20 SECONDS
+	var/mob/living/carbon/human/chosen
+	var/list/possible_chosen = list()
+	var/station_z = level_name_to_num(MAIN_STATION) //idk why GDN removed this, who wants to be a paradox clone of miner or explorer??!?!?!
+
+	for(var/mob/living/carbon/human/H in GLOB.player_list)
+		if(H.mind && H.key && H.z == station_z && H.stat == CONSCIOUS && !istype(get_area(H), /area/station/public/sleep) && H.mind.assigned_role != null && !H.mind.offstation_role && !is_paradox_clone(H))
+			possible_chosen += H
+
+	if(!length(possible_chosen))
+		message_admins("No suitable humans for Paradox Clone event!")
+		abort()
+		return
+
+	while(count)
+		count--
+		chosen = pick(possible_chosen) //not pick n take to have at least very LOW chance to create an interesting situation like one clone need to protect someone, and second clone need to kill them lol
```

Comment:
I think this is better handled with a `for` loop, if you have any questions you can reach out to me!